realm/realm-library/src/main/java/io/realm/internal/Table.java

```diff
@@ -263,6 +263,27 @@ public void renameColumn(long columnIndex, String newName) {
     }
 
     /**
+     * When a {@link Table} is renamed, it is necessary to set the {@link Table#PRIMARY_KEY_CLASS_COLUMN_INDEX}
+     * entry of its primary Key table with a new class name to transfer the primary key.
+     *
+     * @param oldTableName the old table name to search its primary key.
+     * @param newTableName the new table name to reset the primary key
+     */
+    public void resetPrimaryKeyForRenamedClass(String oldTableName, String newTableName) {
```

Comment:
`reset` seems the wrong word here, perhaps "movePrimaryKeyToRenamedClass" instead?